## ASSESSING A DIABETES PREDICTION CLASSIFIER USING  CONFUSION MATRIX

##### Confusion Matrix is a metric used for classification problems when accuracy is not sufficient to reflect the model's true performance.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:

diabetes_df = pd.read_csv('diabetes_clean.csv')

diabetes_df

pregnancies  glucose  diastolic  triceps  insulin   bmi    dpf  age  \
0              6      148         72       35        0  33.6  0.627   50   
1              1       85         66       29        0  26.6  0.351   31   
2              8      183         64        0        0  23.3  0.672   32   
3              1       89         66       23       94  28.1  0.167   21   
4              0      137         40       35      168  43.1  2.288   33   
..           ...      ...        ...      ...      ...   ...    ...  ...   
763           10      101         76       48      180  32.9  0.171   63   
764            2      122         70       27        0  36.8  0.340   27   
765            5      121         72       23      112  26.2  0.245   30   
766            1      126         60        0        0  30.1  0.349   47   
767            1       93         70       31        0  30.4  0.315   23   

     diabetes  
0           1  
1           0  
2           1  
3           0  
4           1  
..        ...  
763         0  
764         0  
765         0  
766         1  
767         0  

[768 rows x 9 columns]

In [4]:
X = diabetes_df.drop('diabetes', axis = 1).values
y = diabetes_df.diabetes

from sklearn.model_selection import train_test_split


X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, stratify = y, random_state = 20)

In [8]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import confusion_matrix, classification_report

In [9]:
knn= KNeighborsClassifier(n_neighbors=6)

knn.fit(X_train, y_train)

y_pred = knn.predict(X_test)

print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

[[87 13]
 [25 29]]
              precision    recall  f1-score   support

           0       0.78      0.87      0.82       100
           1       0.69      0.54      0.60        54

    accuracy                           0.75       154
   macro avg       0.73      0.70      0.71       154
weighted avg       0.75      0.75      0.74       154



##### This shows a 69% precision rate for diabetes, that is, of all the predicted positive values for diabetes, 69% of the were actually correct, while 31% were false positives.

##### The recall rate is 54%, which means that only 54% of actual positive values for diabetes were predicted correctly.